R/wwinference.R

```diff
@@ -189,6 +196,7 @@ wwinference <- function(ww_data,
   )
 
   init_lists <- NULL
+  set.seed(initial_values_seed)
```

Comment:
Use `withr::with_seed` here instead of `set.seed`. See https://withr.r-lib.org/articles/withr.html#its-dangerous-to-change-state
```suggestion
  set.seed(initial_values_seed)
```

Reply:
specifically, change this:

```R
    init_lists <- c()
    for (i in 1:fit_opts$n_chains) {
      init_lists[[i]] <- get_inits_for_one_chain(stan_data_list, params)
    }
```

to this

```R
withr::with_seed(initial_values_seed, {
     init_lists <- lapply(1:fit_opts$n_chains, \(x){get_inits_for_one_chain(stan_data_list, params)}
   })

Reply:
@dylanhmorris I am not following why we would use the :
```r
if(is.null(initial_values_seed)) {
   initial_values_seed <- sample.int(.Machine$integer.max, 1L)
}
```

Isn't the whole point that you want to be able to run :
```r
withr::with_seed(5, {
    fit <- do.call(
      wwinference::wwinference,
      model_test_data
    )
  })
```
And have that seed be passed silently to R, so we'd really want something like:
```r
 if (generate_initial_values) {
     if(is.null(initial_values_seed){

      init_lists <- lapply(
          1:fit_opts$n_chains,
             \(x) {
              get_inits_for_one_chain(stan_data_list)
            }
        )
    }else{
       withr::with_seed(initial_values_seed, {
        init_lists <- lapply(
           1:fit_opts$n_chains,
           \(x) {
             get_inits_for_one_chain(stan_data_list)
          }
        )
      })
   }
```

Reply:
Update: 
 f2f discussion explaining that this:
 ` initial_values_seed <- sample.int(.Machine$integer.max, 1L)` would either be random if no seed set outside of the function or produce the same `initial_values_seed` reproducibly if the seed was set outside of the function, achieving the goal of allowing for both functionalities and stochastic by default